sftplib/src/main/java/com/freiheit/fuava/sftp/RemoteConfigurationWithPlaceholderImpl.java

```diff
@@ -42,14 +43,53 @@
      * @param remoteArchivedFolder location of files have been downloaded successfully from sftp server
      *
      */
-    public RemoteConfigurationWithPlaceholderImpl(final String remoteFilesIncomingFolder, final String remoteProcessingFolder, final String remoteSkippedFolder,
-                                                  final String remoteArchivedFolder) {
+    private RemoteConfigurationWithPlaceholderImpl(
+            final String remoteFilesIncomingFolder,
+            final String remoteProcessingFolder,
+            final String remoteSkippedFolder,
+            final String remoteArchivedFolder,
+            final boolean moveToProcessing ) {
```

Comment:
I would propose to calculate "moveToProcessing" based on whether remoteProcessingFolder is null. This way we can avoid inconsistent states of remoteProcessingDir + moveToProcessing.

Additionally I would propose to add some `Predonditions.checkNotNull` for all folders that should be set.